contrib_bots/lib/github_issues.py

```diff
@@ -0,0 +1,135 @@
+            Also, make sure that the credentials of the github bot have
+            been typed in correctly, that there is a personal access token
+            with access to public repositories ONLY,
+            and that the repository name is entered correctly.
+
+            Check ~.zuliprc-local, and make sure there are
+            github_repo   (The name of the repo to post to)
+            github_repo_owner   (The owner of the repo to post to)
+            github_username   (The username of the github bot)
+            github_token    (The personal access token for the github bot)
+            '''
+
+    def triage_message(self, message):
+        # return True if we want to (possibly) respond to this message
+
+        original_content = message['content']
+
+        # This next line of code is defensive, as we
+        # never want to get into an infinite loop of posting follow
+        # ups for own follow ups!
+        if message['display_recipient'] == 'issue':
+            return False
+        is_issue = (original_content.startswith('@issue'))
+
+        return is_issue
+
+    def handle_message(self, message, client, state_handler):
+        original_content = message['content']
+        original_sender = message['sender_email']
+
+        # gets token from dev-secrets
+        # CONFIGURE TOKEN ON LOCAL MACHINE FOR NOW AT .zuliprc-local
+        config = ConfigParser.ConfigParser()
+        config.read([os.path.expanduser('~/.zuliprc-local')])
+
+        # The repo name to post to
+        REPO_NAME = config.get('api', 'github_repo')
+
+        # The owner of the repo to post to
+        REPO_OWNER = config.get('api', 'github_repo_owner')
+
+        # This adds the sender's username, so that people can be recognised
+        new_content = original_content.replace('@issue',
+                                          'by %s:' % (original_sender,))
+        # gets the repo url
+        url = 'https://api.github.com/repos/%s/%s/issues' % (REPO_OWNER, REPO_NAME)
+
+        # signs into github using the provided username and password
+        session = github.auth()
+
+        # Gets rid of the @issue in the issue title
+        issue_title = message['content'].replace('@issue', '').strip()
+        # Initialises the issue_content variable, the variable that the overflow goes into
+        issue_content = ''
+        # cut_string - true if the string needs to be cut
+        cut_string = False
+        # cut len - the length of the cut string
+        cut_len = 0
+        # iterates for every letter in issue_title
+        for index, character in enumerate(issue_title):
+            # checks to see if there are more than 70 chars, if the string is not already being cut,
+            # and if there is a space (so that it does not cut in the middle of a row)
+            if index > 70 and not cut_string and character == ' ':
+                # If these are furfilled, the string should start to be cut
+                cut_string = True
```

Comment:
can we simplify it to:
`cut_string = index > 70 and not cut_string and character == '  '`?